src/main/java/org/jenkins/plugins/lockableresources/LockStep.java

```diff
@@ -11,25 +11,47 @@
 import hudson.Extension;
 import hudson.model.AutoCompletionCandidates;
 
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+
 public class LockStep extends AbstractStepImpl implements Serializable {
 
-	public final String resource;
+	@CheckForNull
+	public String resource = null;
+
+	@CheckForNull
+	public String label = null;
+
+	public int quantity = 0;
 
 	public boolean inversePrecedence = false;
 
 	@DataBoundConstructor
-	public LockStep(String resource) {
-		if (resource == null || resource.isEmpty()) {
-			throw new IllegalArgumentException("must specify resource");
-		}
-		this.resource = resource;
+	public LockStep() {
```

Comment:
I may be missing something here but I don't think this has to be a breaking change. `resource` can be left in the constructor, and you can use `@DataBoundSetter` for the new fields. Both should work just fine.